pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -399,13 +400,14 @@ func (r *ReconcileCnsNodeVMAttachment) Reconcile(ctx context.Context,
 				msg := fmt.Sprintf("failed to find the VM on VC with UUID: %s for "+
 					"CnsNodeVmAttachment request with name: %q on namespace: %s. Err: %+v",
 					nodeUUID, request.Name, request.Namespace, err)
-				log.Errorf(msg)
-				if err != cnsvsphere.ErrVMNotFound {
+				log.Infof(msg)
+				if strings.Contains(err.Error(), cnsvsphere.ErrVMNotFound.Error()) {
```

Comment:
Can you elaborate your thoughts on changing the logic to check error type here? I'm not sure if one is better than the other.

Reply:
This is not needed. I was explicitly checking for error string just to be sure while testing. Keeping this line as it was before